k256/src/arithmetic/scalar/scalar_8x32.rs

```diff
@@ -207,6 +207,26 @@ impl Scalar8x32 {
         CtOption::new(Self(w), underflow)
     }
 
+    /// Parses the given byte array as a scalar.
+    ///
+    /// Subtracts the modulus when the byte array is larger than the modulus.
+    pub fn from_bytes_reduced(bytes: &[u8; 32]) -> Self {
+        // Interpret the bytes as a big-endian integer w.
+        let w7 = u32::from_be_bytes(bytes[0..4].try_into().unwrap());
+        let w6 = u32::from_be_bytes(bytes[4..8].try_into().unwrap());
+        let w5 = u32::from_be_bytes(bytes[8..12].try_into().unwrap());
+        let w4 = u32::from_be_bytes(bytes[12..16].try_into().unwrap());
+        let w3 = u32::from_be_bytes(bytes[16..20].try_into().unwrap());
+        let w2 = u32::from_be_bytes(bytes[20..24].try_into().unwrap());
+        let w1 = u32::from_be_bytes(bytes[24..28].try_into().unwrap());
+        let w0 = u32::from_be_bytes(bytes[28..32].try_into().unwrap());
+        let w = [w0, w1, w2, w3, w4, w5, w6, w7];
```

Comment:
A bit off-topic, but what is the motivation for this level of explicitness vs. loops or iterator approaches in cryptographic code? When I try this kind of thing, I tend to always get at least one index wrong :)

Reply:
This is actually copypasta from both the `from_bytes` method as well as the `from_overflow` method.

I'd agree it could be simplified and potentially composed in terms of `from_overflow` too.

Reply:
Perhaps some time after https://github.com/rust-lang/rust/pull/74060 is merged this could be:

```rust
let w: [u32; 8] = bytes.chunks(4).map(|chunk| u32::from_be_bytes(chunk).try_into().unwrap()).collect();
```

Reply:
AIUI, in some cases the p256 arithmetic uses "many variables" to allow functions to be const, not in this case though. I was wondering if the use for this was motivated by consistency and subjective preference, or some kind of cryptographic reason, would you know?

Reply:
I don't think there's any specific reason for it, but perhaps @str4d or @fjarri could chime in

Reply:
There is a minor performance benefit as compared to loops, but that can be negated by using an unrolling macro (there are several packages available for that). I have plans to do that, but just didn't get to it yet.

Reply:
Ah, loop unrolling. Thanks!

Reply:
As far as macros for loop unrolling go, I've seen [`unroll`](https://docs.rs/unroll/), but that said so far in my usage of [godbolt](https://godbolt.org/) to optimize cryptographic code I still haven't seen a case where unrolling a loop by hand generates more efficient assembly.

Granted that's been almost exclusively with x86(-64) targets, but...